### ISSUE TITLE
Error while creating LINE social connection

### ISSUE THREAD

**Issue by kikeyama** [auth0/terraform-provider-auth0#891]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

I got the following error when executing `terraform apply` to create `line` connection. LINE connection is actually created in the tenant but the terraform resource is not created.

```
│ Error: failed to unmarshal response payload: json: cannot unmarshal array into Go struct field ConnectionOptionsOAuth2.scope of type string
│
│   with auth0_connection.social_line,
│   on auth0_connection.tf line 54, in resource "auth0_connection" "social_line":
│   54: resource "auth0_connection" "social_line" {
```

### Expectation

It's expected to create LINE social connection and the resource without error message.

### Reproduction

Create the following HCL and execute `terraform apply`.

```
resource "auth0_connection" "social_line" {
  name     = "line"
  strategy = "line"
  options {
    scopes         = ["openid", "profile"]
    client_id     = var.line_client_id
    client_secret = var.line_client_secret
  }
}
```

### Auth0 Terraform Provider version

1.1.1

### Terraform version

1.5.5

**Comment by jalvarez-outburst** [comment]
I have the same issue when creating and amazon connection:
Resource:
```
resource "auth0_connection" "amazon" {
  name     = "amazon"
  strategy = "amazon"

  options {
    client_id     = var.social_amazon_client_id
    client_secret = local.socials_client_secrets.amazon_client_secret
    scopes        = ["profile"]
  }
}
```

Error:
```
│ Error: failed to unmarshal response payload: json: cannot unmarshal array into Go struct field ConnectionOptionsOAuth2.scope of type string
│ 
│   with module.auth0.auth0_connection.amazon,
│   on .terraform/modules/auth0/socials.tf line 1, in resource "auth0_connection" "amazon":
│    1: resource "auth0_connection" "amazon" {
│ 
```

**Comment by developerkunal** [comment]
Greetings everyone,

I hope this message finds you well. Regrettably, we've encountered a delay in addressing an issue that arose due to a change in the API, specifically within the go-auth0 SDK. However, I'm glad to inform you that the problem has been resolved, though the fix hasn't been rolled out officially yet.

Your patience and understanding during this time are greatly appreciated. Should you have any further concerns or encounter any difficulties, please feel free to reach out to us.

Thank you for your ongoing support and cooperation!

**Comment by developerkunal** [comment]
Hi @kikeyama ,

Great news! The issue has been fixed in version 1.2.1. Please take a moment to review, and feel free to reach out if you encounter any further issues.
